src/index.js

```diff
@@ -302,6 +323,55 @@ class PublicationClient extends EventEmitter {
     return subscription;
   }
 
+  /**
+   * Subscribe to the publication via callback parameters. If pressented expected to be called
+   * on receiving the events.
+   *
+   * @param {function} onReconnectedEvent The function that should be called when receiving reconnection.
+   * @param {function} onAddedEvent The function that should be called when receiving added updates.
+   * @param {function} onChangedEvent The function that should be called when receiving changed updates.
+   * @param {function} onRemovedEvent The function that should be called when receiving removed updates.
+   */
+  subscribeEventsToCallbacks(onReconnectedEvent, onAddedEvent, onChangedEvent, onRemovedEvent) {
+    this._onReconnectedEvent = onReconnectedEvent;
+    this._onAddedEvent = onAddedEvent;
+    this._onChangedEvent = onChangedEvent;
+    this._onRemovedEvent = onRemovedEvent;
+    this._subscribedToCallbacks = true;
+  }
+
+  /**
+   * Based on the name and params, gives the key of the subscription
+   *
+   * @param {String} name The publication to subscribe to.
+   * @param {*[]} params (optional) Params to pass to the publication.
+   * @returns {String} The subscription to the desired publication.
+   */
+  getSubscriptionKey() {
+    return JSON.stringify(_.toArray(arguments));
+  }
+  /**
+   * Unubscribe to the publication via callback parameters and for all the given hash name.
+   * @param {{
+   *  name: string - The publication to subscribe to.
+   *  params: *[] - (optional) Params to pass to the publication.
+   * }[]} keys
+   */
+  unsubscribeEventFromCallbacks(keys) {
+    _.map(keys, ({ name, options }) => {
+      const subscriptionKey = options
+        ? this.getSubscriptionKey(name, options)
+        : this.getSubscriptionKey(name);
+      delete this._subscriptions[subscriptionKey];
+    });
+
+    this._onReconnectedEvent = undefined;
+    this._onAddedEvent = undefined;
+    this._onChangedEvent = undefined;
+    this._onRemovedEvent = undefined;
+    this._withoutBackboneCollections = false;
```

Comment:
Shouldn't it be `this._subscribedToCallbacks = false`?

Reply:
Hmmm yes, I lost a commit somewhere thanks :))